src/CanvasRenderingContext2d.cc

```diff
@@ -2073,14 +2073,29 @@ NAN_METHOD(Context2d::MeasureText) {
   String::Utf8Value str(info[0]->ToString());
   Local<Object> obj = Nan::New<Object>();
 
-  PangoRectangle ink_rect, logical_rect;
+  PangoRectangle _ink_rect, _logical_rect;
+  float_rectangle ink_rect, logical_rect;
   PangoFontMetrics *metrics;
   PangoLayout *layout = context->layout();
 
   pango_layout_set_text(layout, *str, -1);
   pango_cairo_update_layout(ctx, layout);
 
-  pango_layout_get_pixel_extents(layout, &ink_rect, &logical_rect);
+  // Normally you could use pango_layout_get_pixel_extents and be done, or use
+  // pango_extents_to_pixels, but both of those round the pixels, so we have to
+  // divide by PANGO_SCALE manually
+  pango_layout_get_extents(layout, &_ink_rect, &_logical_rect);
+
+  logical_rect.x = 1.0 * _logical_rect.x / PANGO_SCALE;
+  logical_rect.y = 1.0 * _logical_rect.y / PANGO_SCALE;
+  logical_rect.width = 1.0 * _logical_rect.width / PANGO_SCALE;
+  logical_rect.height = 1.0 * _logical_rect.height / PANGO_SCALE;
+
+  ink_rect.x = 1.0 * _ink_rect.x / PANGO_SCALE;
+  ink_rect.y = 1.0 * _ink_rect.y / PANGO_SCALE;
+  ink_rect.width = 1.0 * _ink_rect.width / PANGO_SCALE;
+  ink_rect.height = 1.0 * _ink_rect.height / PANGO_SCALE;
```

Comment:
Think it's preferable to explicitly cast one or both operands to a float/double. You could cast `PANGO_SCALE` once and be done.

Reply:
Makes sense, updated PR.